src/components/onlineposts/getonline.js

```diff
@@ -1,138 +1,91 @@
+import "./getonline.css";
 
 class GetOnlinePosts extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      //activePage: 10,
-      pageNumber: 1,
-      //currentPage: 1,
+      currentPageCompleted: 1,
+      currentPageInComplete: 1,
       error: null,
       isLoaded: false,
-      posts: [],
-      // length: 0
+      posts: []
     };
-
   }
 
-
-  changeCurrentPagecomplete = numPage => {
-    this.setState({ currentPage: numPage });
-
+  changeCurrentPageComplete = numPage => {
+    this.setState({ currentPageCompleted: numPage });
   };
-  changeCurrentPageincomplete = numPage => {
-    this.setState({ currentPagePost: numPage });
+  changeCurrentPageInComplete = numPage => {
+    this.setState({ currentPageInComplete: numPage });
   };
 
   componentDidMount() {
     // I will use fake api from jsonplaceholder website
-    // this return 100 posts 
+    // this return 200 posts
     fetch("https://jsonplaceholder.typicode.com/todos")
       .then(response => response.json())
       .then(
         // handle the result
-        (result) => {
-          console.log(result)
-          //iteratees = obj => obj.completed;
-          // const sorted = _.sortBy(result, obj => obj.completed);
-          var partition = _.partition(result, obj => obj.completed)
+        result => {
+          const posts = partition(result, obj => obj.completed);
           this.setState({
             isLoaded: true,
-            posts: partition,
-          })
-
-        })
+            posts
+          });
+        }
+      )
       .catch(error => {
         this.setState({
           isLoaded: true,
           error
-        })
-      }
-      )
+        });
+      });
   }
 
   render() {
-
-    const { error, isLoaded, posts } = this.state;
-    console.log(this.state)
-    // console.log(posts[0])
+    const {
+      error,
+      isLoaded,
+      posts: [completedPosts, inCompletePosts],
+      currentPageCompleted,
+      currentPageInComplete
+    } = this.state;
     if (error) {
-      return <div>Error in loading</div>
+      return <div>Error in loading</div>;
     } else if (!isLoaded) {
-      return <div>Loading ...</div>
+      return <div>Loading ...</div>;
     } else {
-      const posts1 = posts[0];
-      const count1=(this.state.currentPage - 1) * 10
-      const count2=(this.state.currentPagePost - 1) * 10
-      const postscompleted = posts1.slice(count1).slice(0, 10);
-      console.log(this.state.currentPage - 1)
-      console.log("Array Slice" + JSON.stringify(postscompleted));
-      const posts2 = posts[1];
-      const postsincomplete = posts2.slice(count2).slice(0, 10);
-      console.log("Array Slice" + JSON.stringify( postsincomplete));
-      //console.log("Length of the posts" + posts1.length);
+      const completedOffset = (currentPageCompleted - 1) * 10;
+      const inCompleteOffset = (currentPageInComplete - 1) * 10;
+      const activeCompletedPosts = completedPosts.slice(
```

Comment:
I didn't understand a slice method used here??

Reply:
We don't need two `slice` methods. We need the start & end index for the slice.

Reply:
okay